docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6269,6 +6279,314 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
+    different members of different types in GenericData.
+  * generic_bag : Unordered collection of members of same type. Duplicates
+    are allowed.
+  * generic_list : Used to define a list of same types. Only allows for a
+    collection of the same type. It can also be a list of generic_structs.
+    Members are ordered and duplicates are allowed.
+  * generic_set : Used to define a set of same types. It can also be a set
+    of generic_structs. Members are unordered and duplicates are not allowed.
+  * generic_ordered_set : Used to define an ordered set of same types.
+    It can also be a set of generic_structs. Members are ordered and
+    duplicates are not allowed.
+  * string : Used to define control plane strings. The max string length is
+    defined by the size
+  * enum : String representation with which safe enums are realized
+  * enum_val : bytes value with which unsafe/serializable enums are defined
+  * p4_type : This is the same type as P4DataTypeSpec. This helps in achieving
+    parity with P4 types if the GenericTable is being generated from a P4 extern.
+    There are a few overlaps like bool, enums, new_type, bitstring. If the
+    entity is from a P4 extern and a P4 data type exists, then the p4_type is
+    used. Otherwise, the GenericDataType is used
+
+## `GenericTableEntry` { #sec-generic-table-entry}
+See section [GenericTableEntry p4runtime](#sec-p4runtime-generic-table-entry) for more info
+
+GenericTableEntry can be used to program non-PSA externs or non-P4 target-specific
+"fixed" features to their implementation in a generic way. This provides an alternative
+to `Extern` in a structured way and within p4runtime guidelines and semantics.
+The idea has been borrowed from TDI [Table Driven Interface](https://github.com/p4lang/tdi),
+that every state can be representated as a table or multiple tables. A table here is any
+data structure with one or more entries and each entry can be represented as a set of
+key fields and data fields.
+Actions in P4 tables behave like unions and that has been borrowed here in GenericTable.
+This can be potentially used targets to map to either TDI based backends, or even non-TDI
+based backends. If using TDI, then potentially, for every new feature to be added, only
+TDI backend support and remote client code will be required to make use of that function.
+All other middle layers should theoretically remain unchanged.
+
+P4Runtime supports inserting, modifying, deleting and reading GenericTable entries with
+the `GenericTableEntry` entity, which has the following fields:
+
+* `table_id`, which identifies the table instance; the `table_id` is determined
+  by the P4Info message.
+
+* `match`, a repeated field of `GenericFieldMatch` messages. Each element in the
+  repeated field is used to provide a value for the corresponding element in the
+  p4info.
+
+* `union`, which indicates which of the table's unions to execute in case of
```

Comment:
Nit: I have a concern about the use to the word `union` by itself. I understand this is a C union type; but union is also a set operation. When I first read this spec, I wasn't sure if the GenericTableEntry's data could support a set of actions -- more than one action (like OpenFlow).

GenericTableEntry's data is in the form of a variant record. Each GenericTableEntry can execute exactly one variant. I feel that the word `variant` is an improvement on `union` in this context.